File: appli-android/app/src/main/java/com/example/sportapp/core/network/FoodPortionApi.kt
package com.example.sportapp.core.network

import com.example.sportapp.core.data.model.FoodPortion
import retrofit2.http.*

interface FoodPortionApi {

    @GET("food-portions")
    suspend fun getAll(): List<FoodPortion>

    @GET("food-portions/{uuid}")
    suspend fun getByUUID(@Path("uuid") uuid: String): FoodPortion

    @PUT("food-portions/{uuid}")
    suspend fun upsert(@Path("uuid") uuid: String, @Body item: FoodPortion)

    @PUT("food-portions/bulk")
    suspend fun upsertAll(@Body items: List<FoodPortion>)

    @DELETE("food-portions/{uuid}")
    suspend fun delete(@Path("uuid") uuid: String)
}

File: appli-web/src/app/core/api/food-portion-api.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { FoodPortion } from '@core/models/food-portion.model';
import { API_BASE_URL } from './api.config';

@Injectable({ providedIn: 'root' })
export class FoodPortionApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/food-portions`;

  getAll(): Observable<FoodPortion[]> {
    return this.http.get<FoodPortion[]>(this.base);
  }
  upsert(p: FoodPortion): Observable<FoodPortion> {
    return this.http.put<FoodPortion>(`${this.base}/${p.uuid}`, this.toBody(p));
  }
  /** Upsert groupé `PUT /food-portions/bulk` (miroir upsertAll Android) — 1 requête pour N rows. */
  upsertAll(items: FoodPortion[]): Observable<FoodPortion[]> {
    return this.http.put<FoodPortion[]>(`${this.base}/bulk`, items.map((p) => this.toBody(p)));
  }
  delete(uuid: string): Observable<unknown> {
    return this.http.delete(`${this.base}/${uuid}`);
  }

  /** Corps wire attendu par FoodPortionCreate (camelCase). */
  private toBody(p: FoodPortion) {
    return {
      uuid: p.uuid,
      foodUUID: p.foodUUID,
      label: p.label,
      grams: p.grams,
      updatedAt: p.updatedAt,
    };
  }
}
